sql/core/src/test/scala/org/apache/spark/sql/JoinSuite.scala

```diff
@@ -1685,4 +1685,24 @@ class JoinSuite extends QueryTest with SharedSparkSession with AdaptiveSparkPlan
       checkAnswer(sql(query), expected)
     }
   }
+
+  test("SPARK-44132: FULL OUTER JOIN by streamed column name fails with NPE") {
```

Comment:
Did you manually test your original Java reproduction case to ensure that your fix also fixes that case.

Reply:
@bersprockets absolutely.
I also have a [unit test lying around](#41688 ) to validate it, but it feels superfluous to add that one too.  Let me know if you would prefer that I also add/submit it.

Reply:
>Let me know if you would prefer that I also add/submit it.

No, I think the current test is fine. I just wanted to make sure we were testing the original bug.